board/download_dependencies.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+#
+# This script download OpenBLAS dependecy for Armadillo and cross-compile them against
+# required architecture.
```

Comment:
I'd suggest refactoring this script a little bit so that we can use it directly from CMake.  Actually it is not too hard to download a file natively from CMake, and then unpack it too.  You could use `execute_process` for the one instance where you have to build (OpenBLAS).

If you had a separate CMake function or script for each individual download, then you could only download the dependencies that you need.  For instance, just like the ensmallen auto-downloader, we try to use `find_package()`, and if that fails, then we do the download and unpack into `build/deps/`.

In fact, if you got really fancy, you could make a CMake macro that tries to call `find_package()` for a given package, and then downloads it from a given URL if it isn't found.

On top of that, actually downloading dependencies is not specific to cross-compiling.  If we are going through the effort to make an auto-downloader for each of our dependencies, then we can make this available to any user, which could be nice!